dvc/dependency/repo.py

```diff
@@ -209,14 +217,16 @@ def get_meta(self, filter_info=None, **kwargs):
         )
         return meta
 
-    def _make_repo(self, locked=True, **kwargs):
+    def _make_repo(
```

Comment:
I find using `_GeneratorContextManager` here to be a bit weird, an alternative might be doing something like:

```diff
diff --git a/dvc/dependency/repo.py b/dvc/dependency/repo.py
index 3490692b..980c5129 100644
--- a/dvc/dependency/repo.py
+++ b/dvc/dependency/repo.py
@@ -217,14 +217,16 @@ class RepoDependency(Dependency):
         )
         return meta
 
+    @contextmanager
     def _make_repo(
         self, locked: bool = True, **kwargs
-    ) -> "_GeneratorContextManager":
+    ) -> Iterator["Repo"]:
         from dvc.external_repo import external_repo
 
         d = self.def_repo
         rev = self._get_rev(locked=locked)
-        return external_repo(d[self.PARAM_URL], rev=rev, **kwargs)
+        with external_repo(d[self.PARAM_URL], rev=rev, **kwargs) as repo:
+            yield repo
 
     def _get_rev(self, locked: bool = True):
         d = self.def_repo
```

Although it might not be worth it

Reply:
We could type it as `ContextManager[Repo]`. No need to change code, I think.